synapse/storage/databases/main/e2e_room_keys.py

```diff
@@ -301,8 +325,14 @@ async def delete_e2e_room_keys(
                 If not specified, we delete all the keys in this version of
                 the backup (or for the specified room)
         """
+        try:
+            version_int = int(version)
+        except ValueError:
+            # Our versions are all ints so if we can't convert it to an integer,
+            # it doesn't exist.
+            return
```

Comment:
Not sure about this---if someone provides e.g. an empty string for the version and tries to delete stuff with that version, they might think they've deleted something when in fact they've deleted nothing.

But... maybe the point is that this should be handled at the handler/rest levels?

Reply:
The intent of this code was to match the behaviour when a non-existant numeric version is given. `simple_delete` below will always succeed, even if it deletes 0 rows. Only the `_one()` variants will raise a 404 error (which ought to correspond to all the places where I do raise a 404).

Granted, the existing behaviour may be undesirable, but I'd rather not be changing the (intended) API contract as part of this PR.

Reply:
That's very fair, and you're right to call me out for scope creep!